### PR TITLE
Add Support for Windows

### PR DESCRIPTION
**Pull request by ryan-pip**
Add windows compatibility by using  winkerberos.  #28 #22

### CODE REVIEW

File: puresasl/mechanisms.py [hunk 1]
```diff
@@ -13,6 +14,11 @@
 except ImportError:
     have_kerberos = False
 
+if platform.system() == 'Windows' :
+    import winkerberos as kerberos
```
Comment (thobbs) [inline puresasl/mechanisms.py]: Can you wrap this in a `try/except`, like the kerberos import just above this?

File: puresasl/mechanisms.py [hunk 2]
```diff
@@ -13,6 +14,11 @@
...
+    kerberos.authGSSClientUserName = kerberos.authGSSClientUsername
```
Comment (thobbs) [inline puresasl/mechanisms.py]: Is this line performing some magic that I can't figure out, or is it just leftover?

**Comment by thobbs** [comment]
This would be awesome to have! I just have a couple of minor comments on the changes. Thanks!

File: puresasl/mechanisms.py (hunk 1 above)
Comment (ryan-pip) [reply puresasl/mechanisms.py]: Yep, thought of that this morning, ill make that change

File: puresasl/mechanisms.py (hunk 2 above)
Comment (ryan-pip) [reply puresasl/mechanisms.py]: It is doing some magic, the winkerberos library has different capitalisation on the 'N' in name. this just makes it compatible.

Comment (thobbs) [reply puresasl/mechanisms.py]: Ahhh, okay, that makes sense. Can you just add a comment about that? Other than that, I think this is probably good to go.

**Comment by thobbs** [comment]
This looks good to me. @ryan-pip I assume you were able to give this a test run? I haven't done so due to the elaborate setup required.

@ashafer01  do you have any input? If not, I think this is good to merge.

**Comment by ryan-pip** [comment]
Yes, i was able to give it a test run.  Did a run in a clean env using pip to install and was missing a lib in setup.py but fixed now.

**Comment by ashafer01** [comment]
Two very minor suggestions:
* We standardize on using `if "win" in sys.platform` or `platform.system() == 'Windows'`
* We should explicitly perform the capitalization monkey patch in the test module as well rather than letting it happen implicitly by importing the mechanisms module.

👍 with or without these

**Comment by ryan-pip** [comment]
Does anyone have preference one way or another which way we standardise?

If we apply the monkey patch in the test and then removed it in the code wouldn’t we end up with a test that would pass and a module that would fail?

**Comment by ashafer01** [comment]
Good point about the test, yeah let's not do that.

`platform.system()` feels more robust to me but I have no experience with using either solution.

**Comment by thobbs** [comment]
I also prefer `platform.system()`, so I've replaced the check in `setup.py` to standardize on that.

Sounds like we're all on the same page. I'll merge at the end of the day if there are no objections.

**Pull request merged**